index.js

```diff
@@ -3,6 +3,8 @@
 const PluginVirtual = require('./src/lib/plugin')
 const Token = require('./src/util/token')
 
-module.exports = PluginVirtual
+module.exports = PluginVirtual.bind(null, {})
+
+module.exports.MakePluginVirtual = (channel) => (PluginVirtual.bind(null, channel))
```

Comment:
Does this work? Since `PluginVirtual` is a class I think you need to call `new` on it, no?